job-service/src/main/java/com/hpe/caf/services/job/api/JobsPut.java

```diff
@@ -179,24 +179,27 @@ public static String createOrUpdateJob(final String partitionId, String jobId, N
                         .collect(Collectors.toList()));
             }
 
+            final boolean partitionSuspended = ApiServiceUtil.isPartitionSuspended(config.getSuspendedPartitionsPattern(), partitionId);
             //  Create job in the database.
-            LOG.info("createOrUpdateJob: Creating job in the database...");
+            LOG.info("createOrUpdateJob: Creating job in the database for {} : {}...",
+                    partitionSuspended ? "suspended partition" : "partition", partitionId);
             final boolean jobCreated;
-            if (job.getPrerequisiteJobIds() != null && !job.getPrerequisiteJobIds().isEmpty()) {
+            if ((job.getPrerequisiteJobIds() != null && !job.getPrerequisiteJobIds().isEmpty()) || partitionSuspended) {
                 jobCreated = databaseHelper.createJobWithDependencies(partitionId, jobId, job.getName(), job.getDescription(),
                         job.getExternalData(), jobHash, jobTask.getTaskClassifier(), jobTask.getTaskApiVersion(),
                         getTaskDataBytes(jobTask, codec), jobTask.getTaskPipe(), jobTask.getTargetPipe(),
-                        job.getPrerequisiteJobIds(), job.getDelay(), job.getLabels());
+                        job.getPrerequisiteJobIds(), job.getDelay(), job.getLabels(), partitionSuspended);
 
             } else {
-                jobCreated = databaseHelper.createJob(partitionId, jobId, job.getName(), job.getDescription(), job.getExternalData(), jobHash, job.getLabels());
+                jobCreated = databaseHelper.createJob(partitionId, jobId, job.getName(), job.getDescription(),
+                        job.getExternalData(), jobHash, job.getLabels(), partitionSuspended);
             }
 
             if (!jobCreated) {
                 return "update";
             }
 
-            if (!databaseHelper.canJobBeProgressed(partitionId, jobId)) {
+            if (!databaseHelper.canJobBeProgressed(partitionId, jobId) || partitionSuspended) {
```

Comment:
Is this change required?  Won't `canJobBeProgressed()` return `false` if the partition is suspended?